src/main/java/com/kobylynskyi/graphql/codegen/mapper/GraphQLTypeMapper.java

```diff
@@ -285,10 +326,13 @@ default NamedDefinition getLanguageType(MappingContext mappingContext, String gr
      * @param parentTypeName  Name of the parent type
      * @param directives      List of GraphQL directive
      * @param mandatory       Type is mandatory
+     * @param isUnion         Is it union type
+     * @param typeName        Name of GraphQL type, but only not null when type is Enum
      * @return list of Java annotations for a given GraphQL type
      */
     default List<String> getAnnotations(MappingContext mappingContext, String graphQLTypeName, String name,
```

Comment:
@gdhordain @jxnu-liguobin Instead of providing 2 extra parameters I'd recommend passing the type itself, so that required information can be fetched in the place it is needed. For example, you can pass either `Type<?>`, or `ExtendedDefinition<?, ?>` or `NamedDefinition` which then can be used in: 
* `ScalaGraphQLTypeMapper` (the only place `typeName` is used)
* `getJacksonTypeIdAnnotations` (the only place where `isUnion` is used)

Another benefit from passing a single type parameter is that in the future we might need to fetch another type attribute directly in the method instead of extending this method signature.

Please let me know what you think.
Hope it make sense to you.

Reply:
It's good.

Reply:
I makes sense, I'll do it.

Reply:
I pushed a refactor which use the definition.
Is it ok for you ?